apps/estuary/lib/estuary/init_server.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Estuary.InitServer do
+  @moduledoc """
+  Task to initialize estuary and start ingesting each previously recorded dataset
+  """
+  use GenServer
+
+  alias Estuary.DataReader
+  alias Estuary.Datasets.DatasetSchema
+  alias Estuary.DataWriter
+
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, [], name: name)
+  end
+
+  def init(_) do
+    with :ok <- create_table(),
+         :ok <- DataReader.init(),
+         pid <- Process.whereis(Pipeline.DynamicSupervisor) do
```

Comment:
I didn't realize that Process.whereis could return something other than a pid, but both port and nil make a lot of sense as alternatives. I see an example in Valkyrie, though it looks like forklift also needs a guard.

I'd like to better understand what we're doing here. It seems like we want to get a pid so when we get a down message we can match against our pid and restart if needed. That being said, wouldn't a guard clause mean we'd kill our init process if we got a port back (because the supervisor was on another node etc)? I guess that's fine since we NEED the exact pid, I just feel like I could be missing some pieces.